Tasks/JavaToolInstaller/javatoolinstaller.ts

```diff
@@ -0,0 +1,101 @@
+import taskLib = require('vsts-task-lib/task');
+import toolLib = require('vsts-task-tool-lib/tool');
+import path = require('path');
+
+import { AzureStorageArtifactDownloader } from "./AzureStorageArtifacts/AzureStorageArtifactDownloader";
+import { JavaFilesExtractor } from "./FileExtractor/JavaFilesExtractor";
+taskLib.setResourcePath(path.join(__dirname, 'task.json'));
+
+async function run() {
+    try {
+        let versionSpec = taskLib.getInput('versionSpec', true);
+        await getJava(versionSpec);
+    }
+    catch (error) {
+        console.error('ERR:' + error.message);
+    } 
+}
+
+async function getJava(versionSpec: string) {
+    toolLib.debug('Trying to get tool from local cache');
+    let localVersions: string[] = toolLib.findLocalToolVersions('Java');
+    let version: string = toolLib.evaluateVersions(localVersions, versionSpec);
+    let fromAzure: boolean = ('AzureStorage' == taskLib.getInput('jdkSource', true));
+    let fromLocalDirectory: boolean = ('LocalDirectory' == taskLib.getInput('jdkSource', true));
+    let extractLocation: string = taskLib.getPathInput('destinationFolder', true);
+    let fileName: string;
+    let cleanDestinationFolder: boolean = taskLib.getBoolInput('cleanDestinationFolder', false);
+    let fileEnding: string;
+
+     // Clean the destination folder before downloading and extracting?
+     if (cleanDestinationFolder && taskLib.exist(extractLocation)) {
+        console.log(taskLib.loc('CleanDestDir', extractLocation));
+        taskLib.rmRF(extractLocation);
```

Comment:
This will attempt to delete the directory right? We should only try to delete the contents of the destination directory since the option says "clean". @lkillgore did this for another task recently